src/posix/platform/trel_udp6.cpp

```diff
@@ -143,7 +147,12 @@ static void AddUnicastAddress(const otIp6Address *aUnicastAddress)
 
     ret = ioctl(mgmtFd, SIOCSIFADDR, &ifr6);
 
-    VerifyOrDie((ret == 0) || (errno == EALREADY) || (errno == EEXIST), OT_EXIT_ERROR_ERRNO);
+    if (!(ret == 0 || errno == EALREADY || errno == EEXIST))
+    {
+        otLogCritPlat("[trel-plat] Failed to add unicast address %s on TREL netif \"%s\"",
+                      Ip6AddrToString(aUnicastAddress), sInterfaceName);
```

Comment:
```suggestion
        otLogCritPlat("[trel-plat] Failed to add unicast address %s on TREL netif \"%s\": %s",
                      Ip6AddrToString(aUnicastAddress), sInterfaceName, strerror(errno));
```

it whould be helpful for us to find out why it fails.

Reply:
The `errno` status is logged from the `DieNow()` which can give more info.